### PR TITLE
Filter out GcnEvents with a specific tag

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR enables filtering out GcnEvents with a specific tag.

### CODE REVIEW

File: skyportal/handlers/api/gcn.py [hunk 1]
```diff
@@ -450,12 +450,19 @@ async def get(self, dateobs=None):
...
                 type: string
               description: |
                 Gcn Tag to match against
+            - in: query
+              name: tagRemove
+              nullable: true
+              schema:
+                type: string
+              description: |
+                Gcn Tag to filter out
```
Comment (guynir42) [inline skyportal/handlers/api/gcn.py:465]: We should do something similar to this with source classifications

File: skyportal/handlers/api/gcn.py [hunk 2]
```diff
@@ -547,15 +555,24 @@ async def get(self, dateobs=None):
...
                 )
+            if tag_remove:
+                tag_subquery = (
+                    GcnTag.select(session.user_or_token)
+                    .where(GcnTag.text.contains(tag_remove))
+                    .subquery()
+                )
+                query = query.join(
+                    tag_subquery, GcnEvent.dateobs != tag_subquery.c.dateobs
+                )
```
Comment (guynir42) [inline skyportal/handlers/api/gcn.py:575]: Could this work with just `.where(~GcnTag.text.contains(tag_remove))`?

**Review by guynir42 — APPROVED** [review]
Minor comment below. I am also assuming here that you've actually filtered on GCNs using the new front end keep/reject form options.

File: skyportal/handlers/api/gcn.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:575]: I did try that but it was confused!

File: skyportal/handlers/api/gcn.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:465]: yeah maybe one of the undergrads can merge the logic in a future PR.

**Pull request merged**